### PR TITLE
BUGZ-993: Bump minimum threadpool size to 2

### PR DESCRIPTION
**Pull request by SimonWalton-HiFi**
Jira: https://highfidelity.atlassian.net/browse/BUGZ-993

Related to: https://highfidelity.atlassian.net/browse/BUGZ-918

`Application::updateThreadPoolCount()` tries to reserve some threads away from the global QThreadPool but in some cases this has been observed to reduce the pool to the minimum of 1. To avoid task starvation it's probably better to have a larger number of threads and let the OS worry about it.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20360/HighFidelity-Beta-PR15922-33494d7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20360/HighFidelity-Beta-Interface-PR15922-33494d7.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20360/HighFidelity-Beta-PR15922-33494d7.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20360/HighFidelity-Quest-Beta-PR15922-33494d7.apk)

**Comment by howard-stearns** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20366/HighFidelity-Beta-PR15922-33494d7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20366/HighFidelity-Beta-Interface-PR15922-33494d7.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20366/HighFidelity-Beta-PR15922-33494d7.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20366/HighFidelity-Quest-Beta-PR15922-33494d7.apk)

**Comment by SimonWalton-HiFi** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20376/HighFidelity-Beta-PR15922-33494d7.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20376/HighFidelity-Quest-Beta-PR15922-33494d7.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20376/HighFidelity-Beta-PR15922-33494d7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20376/HighFidelity-Beta-Interface-PR15922-33494d7.dmg

**Comment by SimonWalton-HiFi** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20389/HighFidelity-Beta-PR15922-33494d7.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20389/HighFidelity-Quest-Beta-PR15922-33494d7.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20389/HighFidelity-Beta-PR15922-33494d7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20389/HighFidelity-Beta-Interface-PR15922-33494d7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20389/HighFidelity-Beta-PR15922-33494d7.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20389/HighFidelity-Beta-Interface-PR15922-33494d7.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20776/HighFidelity-Beta-PR15922-baa5869.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20776/HighFidelity-Quest-Beta-PR15922-baa5869.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20776/HighFidelity-Beta-PR15922-baa5869.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20776/HighFidelity-Beta-Interface-PR15922-baa5869.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20776/HighFidelity-Beta-PR15922-baa5869.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20776/HighFidelity-Beta-Interface-PR15922-baa5869.dmg

**Comment by SimonWalton-HiFi** [comment]
Needs pre-merge testing on HMD and low-end (i3 or i5) platforms.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20777/HighFidelity-Beta-PR15922-0f6ad43.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20777/HighFidelity-Quest-Beta-PR15922-0f6ad43.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20777/HighFidelity-Beta-PR15922-0f6ad43.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20777/HighFidelity-Beta-Interface-PR15922-0f6ad43.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20777/HighFidelity-Beta-PR15922-0f6ad43.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20777/HighFidelity-Beta-Interface-PR15922-0f6ad43.dmg

File: plugins/oculus/src/OculusBaseDisplayPlugin.h [hunk 1]
```diff
@@ -38,6 +38,8 @@ class OculusBaseDisplayPlugin : public HmdDisplayPlugin {
 
     virtual StencilMaskMode getStencilMaskMode() const override { return StencilMaskMode::MESH; }
     virtual StencilMaskMeshOperator getStencilMaskMeshOperator() override;
+    // Attempt to reserve two threads.
+    int getRequiredThreadCount() const override { return 2; }
```
Comment (jherico) [inline plugins/oculus/src/OculusBaseDisplayPlugin.h]: you should move this to `HmdDisplayPlugin`, so it applies to all the HMD plugins.

Comment (SimonWalton-HiFi) [reply plugins/oculus/src/OculusBaseDisplayPlugin.h]: I'll move it. Note that `OpenVrDisplayPlugin` already overrides `getRequiredThreadCount()`.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20800/HighFidelity-Beta-PR15922-99e1178.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20800/HighFidelity-Beta-Interface-PR15922-99e1178.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20800/HighFidelity-Beta-PR15922-99e1178.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20800/HighFidelity-Beta-Interface-PR15922-99e1178.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20800/HighFidelity-Beta-PR15922-99e1178.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20800/HighFidelity-Quest-Beta-PR15922-99e1178.apk)

File: plugins/oculus/src/OculusBaseDisplayPlugin.h (hunk 1 above)
Comment (jherico) [reply plugins/oculus/src/OculusBaseDisplayPlugin.h]: Granted, but it's implemented in terms of the parent class version.

**Comment by antoninahifi** [comment]
Pre merge testing passed on

Toni's: MacBook Pro    i5 2.3    intel iris 640 1536mb    8gb     2560x1600 (intel iris)  , 2 cores
Peter's (Low-end):
- Intel(R) Core(TM) i5-6400 CPU @ 2.70GHz, 2701 Mhz, 4 Core(s), 4 Logical Processor(s)
- AMD Radeon R9 200 Series
Evan's (Occulus):
- AMD Ryzen 2700
- Nvidia 2070

**Pull request merged**